tripy/tests/integration/test_stack.py

```diff
@@ -0,0 +1,62 @@
+#
+# SPDX-FileCopyrightText: Copyright (c) 1993-2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+import cupy as cp
+import pytest
+import tripy as tp
+
+from tests.helper import raises
+
+
+class TestStack:
+    @pytest.mark.parametrize(
+        "tensor_shapes, dim",
+        [
+            ([(2, 3, 4), (2, 3, 4)], 0),
+            ([(2, 3, 4), (2, 3, 4)], 1),
+            ([(2, 3, 4), (2, 3, 4)], -1),
+            ([(2, 3, 4)], 0),
```

Comment:
Do we need to cover the cases where:
1. unsqueezing is required
2. The stack axis has different lengths in the inputs

Reply:
1. `unsqueeze` is always required.
2. stack operates via adding a new dimension to all tensors and concatenating along them. You do not stack on existing axis